File: app/src/utils/imageGenerators/thumbSizeEnum.py
from enum import Enum, unique


@unique
## This enum describes the different formats of the thumbnails.
class ThumbSizeEnum(Enum):
    ## A tiny thumbnail (80x80). Used on the track preview for example.
    TINY = (100, 100)
    ## A small thumbnail (140x140). Used on the main page for example.
    SMALL = (140, 140)
    ## A normal thumbnail (285x285). Used on the detail sceneviews.
    MEDIUM = (300, 300)
    ## A large thumbnail (400x400). Used on the detail sceneviews.
    LARGE = (425, 425)

File: back/src/main/java/org/manazeak/manazeak/constant/file/ThumbSizeEnum.java
package org.manazeak.manazeak.constant.file;

/**
 * Contains the different size of the thumbnails.
 */
public enum ThumbSizeEnum {

    TINY("tiny", 50, 50),
    SMALL("small", 128, 128),
    MEDIUM("medium", 256, 256),
    LARGE("large", 512, 512),
    ORIGINAL("orig", 0, 0);

    private final String folderName;

    private final int height;

    private final int width;

    ThumbSizeEnum(String folderName, int height, int width) {
        this.folderName = folderName;
        this.height = height;
        this.width = width;
    }

    public String getFolderName() {
        return folderName;
    }

    public int getHeight() {
        return height;
    }

    public int getWidth() {
        return width;
    }
}
